### DISCUSSION TITLE
Projects with multiple cpp2 files

### DISCUSSION THREAD

**Discussion by AdRiley** [hsutter/cppfront#1344] [Q&A]
Sorry this is a newbie question, but what is the correct way of splitting a project across multiple .cpp2 files?

Say I have this code

main: () =
{
Hello();
}

Hello : () =
{
std::cout << "Hello World";
}

And want to move the definition of Hello into a different file?

**Comment by JohelEGP** [comment]
I think this is a subject of #594.

**Reply by AdRiley** [reply]
Yes I read that one, but it seems more about using headers to work alongside cpp code. What do I do in a pure ccp2 project?

**Reply by JohelEGP** [reply]
You still need Cpp2 headers, unless you're OK with forward-declaring `Hello` in Cpp1.

**Reply by zaucy** [reply]
I think for pure cpp2 projects we should push towards using modules exclusively, but cppfront doesn't support that (yet) #269

**Reply by threeifbyair** [reply]
I'm doing a multiple-cpp2 project right now and using Johel's modules version of cppfront. It's working pretty well. https://github.com/threeifbyair/advent2024
